presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -241,15 +242,10 @@ PlanNodeStatsEstimate calculateJoinComplementStats(
             PlanNodeStatsEstimate leftStats,
             PlanNodeStatsEstimate rightStats)
     {
-        // TODO: add support for non-equality conditions (e.g: <=, !=, >)
-        if (filter.isPresent()) {
-            // non-equi filters are not supported
-            return UNKNOWN_STATS;
-        }
-
         if (criteria.isEmpty()) {
             if (rightStats.getOutputRowCount() > 0) {
                 // all left side rows are matched
+                // TODO: account non-equi conditions here somehow
                 return leftStats.mapOutputRowCount(rowCount -> 0.0);
```

Comment:
This was correct based on the assumption that `! filter.isPresent()`, which is not necessarily true anymore.

Reply:
It looks here should be some if with `! filter.isPresent()` or logic from line 322 used.

Can you add coverage this case where there is no equi criteria, but filter present?